File: src/optic/lua/ssa/instructions/Return.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

import java.util.List;

public final class Return implements Step {
	private final List<Register> registers;

	public Return(List<Register> registers) {
		this.registers = registers;
	}

	@Override
	public String toString() {
		return "return " + registers;
	}
}

File: src/optic/lua/asm/instructions/Return.kt
package optic.lua.asm.instructions

import optic.lua.asm.RValue
import optic.lua.asm.Step
import optic.lua.asm.StepVisitor
import java.util.function.Consumer

class Return(val values: List<RValue>) : Step {
    override fun <T : Any, X : Throwable> accept(visitor: StepVisitor<T, X>): T = visitor.visitReturn(values)

    override fun toString(): String {
        return "return $values"
    }

    override fun forEachObserved(action: Consumer<RValue>) = values.forEach(action)
}
